ONSdigital/census-rm-toolbox
# 1443 automated bulk file processing

# Checklist
Reminder: If any change made to this repo affects the acceptance tests then the version pin in the Pipfile in [census-rm-acceptance-tests]() needs to be updated.

# Motivation and Context

We're gonna automate the bulk processors to save time during Census. While doing this we've updated the logging to json formatting so now it can be picked up by our GCP metrics.

# What has changed

- Added script to automate bulk processors
- Updated logging

# How to test?




- Build a toolbox image from the PR and push it to your local GCR.
- Deploy the new cronjob bulk processor in the kubernetes PR
- The cronjob should run the automate_bulk_processor script in toolbox on the hour every hour
- Put a couple of files in the bulk buckets and the script should process them all successfully
- Throw in some bad messages and you can see the failures be picked up metrics
- Make the refusal bulk processor fail and it'll show some error lines in the script but carry on the through the rest of the bulk processor.

# Links



[Trello](https://trello.com/c/eXayJFQQ/)
